DirectProgramming/Fortran/DenseLinearAlgebra/optimize-integral/README.md

```diff
@@ -0,0 +1,194 @@
+     make 
+
+   * Run the program
+  
+     make run
+
+   * Note the final run time (example)
+    CPU Time = 3.776983 seconds
+
+   * Clean the files we built
+   
+    make clean
+
+
+### STEP 2: Build and run with -O1
+Edit 'Makefile' using your favorite editor
+
+To set optimization level uncomment FC = ifort -O1 like this
+    
+     #FC = ifort -O0
+     FC = ifort -O1
+     #FC = ifort -O2
+     #FC = ifort -O3
+   * Build the executable with 'make'
+  
+    make
+
+   * Run the program
+  
+    make run
+
+   * Note the final run time (example)
+    CPU Time = 1.444569 seconds
+
+   * Clean the files we built
+    
+    make clean
+    
+
+### STEP 3: Build and run with -O2
+Edit 'Makefile' using your favorite editor
+
+To set optimization level uncomment FC = ifort -O2 like this
+    
+     #FC = ifort -O0
+     #FC = ifort -O1
+     FC = ifort -O2
+     #FC = ifort -O3
+   * Build the executable with 'make'
+   
+    make
+
+   * Run the program
+    
+    make run
+
+   * Note the final run time (example)
+     CPU Time = 0.5143980 seconds
+
+   * Clean the files we built
+  
+    make clean
+
+### STEP 4: Build and run with -O3
+Edit 'Makefile' using your favorite editor
+
+To set optimization level uncomment FC = ifort -O3 like this
+
+     #FC = ifort -O0
+     #FC = ifort -O1
+     #FC = ifort -O2
+     FC = ifort -O3
+   * Build the executable with 'make'
+    
+    make
+
+   * Run the program
+    
+    make run
+
+   * Note the final run time (example)
+     CPU Time = 0.5133380 seconds
+
+   * Clean the files we built
+   
+    make clean
+
+## What did we learn?
+There are big jumps going from O0 to O1, and from O1 to O2. 
+But we see very little performance gain going from O2 to O3.
+This does vary by application but generally with Intel® Compilers 
+O2 is has most optimizations.  Sometimes O3 can help, of course,
+but generally O2 is sufficient for most applications. 
+
+### Further Exploration
+The Intel® Fortran Compiler has many options for optimization. 
+If you have a genuine Intel® Architecture processor, try these additional options
+
+   edit 'Makefile' using your favorite editor. To set additional optimizations uncomment FC = ifort -O3 and add additional options shown:
+   
+     #FC = ifort -O0
+     #FC = ifort -O1
+     #FC = ifort -O2
+     FC = ifort -O3 -xhost -align array64byte
+   * Build the executable with the new options -xhost -align array64byte
+  
+    make
+
+   * Run the program
+   
+    make run
+
+   * Note the final run time (example)
+     CPU Time = 0.2578490 seconds
+
+   * Clean the program
+
+    make clean
+    
+There are 2 additional compiler options here that are worth mentioning: Read the online 
+[Developer Guide and Reference][3] for more information about
+these options
+[3]: https://software.intel.com/content/www/us/en/develop/documentation/fortran-compiler-developer-guide-and-reference/top.html "Developer Guide and Reference" 
+ 1. -xhost (sub option of -x option):  [-x][4]
```

Comment:
formatting issue, why is this red in color

Reply:
it's not in red.  Your viewer must me mistaken.  macDown editor shows markdown source and rendered version just fine for me - there is no markup for RED that I know of and certainly none in the source file.  Check in a text editor.